lib/perl/Genome/Model/ClinSeq/Command/UpdateAnalysis.pm

```diff
@@ -820,12 +839,42 @@ sub get_trsn{
 
   my %trsns;
   my $trsn_ref;
-
-  foreach my $instrument_data (@instrument_data){
-    my $trsn = $instrument_data->target_region_set_name;
-    if ($trsn){
-      $trsns{$trsn}=1;
-      $trsn_ref = $trsn;
+  
+  #When using validation-as-exome and my-trsn options, use user-defined target region set name (1=exome or 2=validation)
+  if ($self->validation_as_exome && $self->my_trsn) {
+    my $user_trsn = $self->my_trsn;
+    
+    #Check if a valid option for my-trsn and use or warn user and exit
+    if ($user_trsn == 1){
```

Comment:
Great suggestions. I've made a commit with these improvements. I didn't realize I could constrain the text values in that way. Thanks!